app/assets/javascripts/discourse/app/components/sidebar/user/categories-section.js

```diff
@@ -5,24 +5,40 @@ import { cached } from "@glimmer/tracking";
 
 import SidebarCommonCategoriesSection from "discourse/components/sidebar/common/categories-section";
 
+export const REFRESH_COUNTS_APP_EVENT_NAME =
+  "sidebar:refresh-categories-section-counts";
+
 export default class SidebarUserCategoriesSection extends SidebarCommonCategoriesSection {
   @service router;
   @service currentUser;
+  @service appEvents;
 
   constructor() {
     super(...arguments);
 
     this.callbackId = this.topicTrackingState.onStateChange(() => {
-      this.sectionLinks.forEach((sectionLink) => {
-        sectionLink.refreshCounts();
-      });
+      this.appEvents.trigger(REFRESH_COUNTS_APP_EVENT_NAME);
```

Comment:
Just to make sure I understand this correctly: this indirection is necessary for plugin API, right?

Reply:
Yes this is correct. The plugin will be triggering a refrehs.